app/store/aliases.ts

```diff
@@ -0,0 +1,28 @@
+import * as axios from "axios";
+import {fetchSuccess, fetchFailure, handleRespTextChange} from "./../actions";
+
+const fetchDataAlias = (payload: object) => {
+  return dispatch => {
+    const {method, url, requestId, requestHeaders} = payload.requestDetails;
+    const requestHeadersObject = requestHeaders.reduce((accumulatedObj, reqHeaderNameValuePair) => {
+      accumulatedObj[reqHeaderNameValuePair.name] = reqHeaderNameValuePair.value;
+      return accumulatedObj;
+    }, {});
+    axios({
+      method,
+      url,
+      requestHeadersObject
+    })
+      .then(data => {
+        dispatch(fetchSuccess(data, requestId));
+        dispatch(handleRespTextChange(data.data, requestId))
+      })
+      .catch(err => {
```

Comment:
try to avoid `catch` when not required. The second argument to `then` here has the same affect as what catch does. using a `.catch` handler would mean that any JS errors or other errors also get caught. Not a huge problem here, but that can become a habit

Reply:
The `catch` block here is executed in case the `axios` lib isn't able to complete and fetch the response.  Don't you recoemmend using this in case the server returns a error such as `404` or `500`. Refer [this](https://github.com/axios/axios#handling-errors)